src/common/registry.rs

```diff
@@ -185,7 +185,61 @@ impl ComponentRegistry {
         if let Some(ctor) = self.encoders.get(model) {
             return Ok(*ctor);
         }
-        Err(RegistryError::ModelNotFound)
+        Err(RegistryError::ModelNotFound(model.to_string()))
+    }
+
+    pub(crate) fn get_motor_constructor_dyn(
```

Comment:
are all these `_dyn` variations only different in that the `model` parameter is a String and not a slice?
If so, you should be able to pass a reference to the strings when calling the non-`dyn` versions. 

edit: I see now that they are taking static strings... might complicate things. Do we even need the `'static`?

Reply:
would add a lifetime <'a> to the function solve the issue?

Reply:
I have a different refactor in mind, which I've pushed up in [this commit](https://github.com/viamrobotics/micro-rdk/pull/50/commits/16749aa2afa4aa6114028ac08a1b1b82a6b1b620). I'd like to know what you both think.